torch/_inductor/codegen/triton_foreach.py

```diff
@@ -144,10 +153,11 @@ def create_sub_kernel(self, *groups, index_dtype, mutations, reduction_hint):
 
     def jit_line(self):
         can_use_32bit = all(k.index_dtype == "tl.int32" for k in self.sub_kernels)
+        size_dtype = "tl.int32" if can_use_32bit else "tl.int64"
         index_dtype = "tl.int32" if can_use_32bit else "tl.int64"
         _, _, signature = self.args.python_argdefs()
         triton_meta = {
-            "signature": signature_to_meta(signature, size_dtype=can_use_32bit),
```

Comment:
mind vetting this part? `signature_to_meta` doesn't take a bool; it takes a type as a string.

Reply:
I believe you can remove index_dtype and just use the size_dtype that you added, the values should be the same